### DISCUSSION TITLE
Support save request/response body in self defined nginx variables

### DISCUSSION THREAD

**Discussion by wangrzneu** [Kong/kong#7237] [Ideas and feature requests]
### Summary

Support save request/response body in self defined nginx variables, then the request/response body content can be saved in access log.

If you think it is a good feature. Let me have a try.

**Comment by bungle** [comment]
There are two options:
1) you grap the body in context variable with `body_filter` phase (that can be called multiple times)
2) implement response phase or call kong.service.request.enable_buffering() with plugin and use kong.service.request_get_raw_body(), but this only works when client connection is http 1.1 due to OpenResty limitation with subrequests.
